### DISCUSSION TITLE
Does the behavior of urls from Rich markdown vary across terminals .?

### DISCUSSION THREAD

**Discussion by vigne-Sh** [Textualize/textual#3707] [Q&A] [answered]
Hello Team,

I wanted to discuss the feasibility of this particular use-case

Rendering url links inside a table using `rich_markdown` is varying across different terminals

* Tried to add urls inside the rich table, but the behavior is varying across different terminals

* Additionally came across the [`Markdown.LinkClicked`](https://github.com/Textualize/textual/discussions/3668), it seems to be  working in all the terminals 

### Linux ( Doesn't work )

* xterm - v372
* alacritty - v0.12.3
* kitty - v0.21.2
* countour (and it's the closest I could find to iTerm2)- v0.3.12.262

### Mac OSX

* iterm - v3.4.22 ( works when pressing the url link with  `command` key )
* default mac terminal - doesn't work 

```
from textual.app import App, ComposeResult
from textual.containers import *
from textual.widgets import *
import webbrowser
from rich.markdown import Markdown as rich_md
from rich.table import Table
table = Table(title="test data")
table.add_column("col1")
table.add_column("col2")
table.add_row(rich_md("[test-link](https://www.google.com)"), '2')

MESSAGE = """
[Test link](https://www.google.com)
"""


class TestUrlApp(App):
    def on_markdown_link_clicked(self, event: Markdown.LinkClicked) -> None:
        webbrowser.open(event.href)

    def compose(self) -> ComposeResult:
        yield Header()
        yield Static(table)
        yield Markdown(MESSAGE)
        yield Footer()


app = TestUrlApp()

if __name__ == "__main__":
    app.run()
```

* Is there any other alternate way to make the url's clickable when its rendered from rich markdown inside the table ?

**Comment by TomJGooding** [comment, marked as answer]
These type of links created with Rich link markup aren't supported everywhere, as discussed recently https://github.com/Textualize/textual/issues/3685#issuecomment-1813027702

However you can achieve this using Textual's [action Links](https://textual.textualize.io/guide/actions/#links).

This is what is used internally by Textual's `Markdown` widget. Is there a reason you're using Rich markdown instead?

[EDIT: Sorry ignore my previous post,  I clearly didn't read your code example very closely!]

**Comment by vigne-Sh** [comment]
Thanks @TomJGooding  for pointing it out, was able to achieve the use case with `action links` 

* Adding the updated example here for any one to refer

### Use-case

* Add url links inside table
* Update the table in frequent time intervals with latest service call data of new urls

```
import webbrowser

from textual.app import App, ComposeResult
from textual.widgets import DataTable
from textual import work

table_cols = ("url_col", "test_value")

ROWS = [
    ("[@click=link('https://github.com/Textualize/textual')]Textual[/]", 22_031),
    ("[@click=link('https://github.com/Textualize/rich')]Rich[/]", 45_483),
]

# func call mock data - which updates new row urls each time when called
Updated_ROWS = [
    ("[@click=link('https://textual.textualize.io/widgets/static/#textual.widgets._static.Static.update')]Static[/]", 0o1),
    ("[@click=link('https://textual.textualize.io/widgets/data_table/#textual.widgets._data_table.DataTable.remove_column')]DataTable[/]", 0o07),
]


class ExampleApp(App):
    def compose(self) -> ComposeResult:
        yield DataTable(id="one")
        self.set_interval(7, self.refresh_first_set_of_tabs)

    def refresh_first_set_of_tabs (self) -> None:
        """
        Refresh First set of tabs in the background
        by querying & update the widgets
        """
        self.load_table_for(
            "one",
            refresh_it=True
        )


    def on_mount(self) -> None:
        self.load_table_for(
            "one",
        )

    def action_link(self, href: str) -> None:
        webbrowser.open(href)

    @work(thread=True)
    def load_table_for (
        self, table_name: str, refresh_it=False
    ) -> None:
        """
        load tables from various services in the dashboard when starting
        """
        table = self.query_one(f"#{table_name}", DataTable)
        if not refresh_it:
            table.add_columns(*table_cols)
            table.add_rows(ROWS)
        if refresh_it:
            table.clear()
            # makes func call & updates latest data
            table.add_rows(Updated_ROWS)

if __name__ == "__main__":
    app = ExampleApp()
    app.run()
```
